specs/icmp.rflx

```diff
@@ -1,21 +1,137 @@
+	  	Echo_Request => 8,
+	  	Dest_Unreacheable => 3,
+	  	Time_Exceeded => 11,
+	  	Parameter_Problem => 12,
+	  	Source_Quench => 4,
+	  	Redirect => 5,
+	  	Timestamp_Msg => 13,
+	  	Timestamp_Reply => 14,
+	  	Information_Request => 15,
+	  	Information_Reply => 16
+	) with Size => 8;
+
+	type CodeDestUnreachable is (
+		Net_Unreachable => 0,
+		Host_Unreachable => 1,
+		Protocol_Unreachable => 2,
+		Port_Unreachable => 3,
+		Fragmentation_Needed_DF_Set => 4,
+		Source_Route_Failed => 5
+	) with Size => 8;
+
+	type CodeTimeExceeded is (
+		TTL_Exceeded => 0,
+		Fragment_Reassembly_Time_Exceeded => 1
+	) with Size => 8;
+
+	type CodeRedirect is (
+		Redirect_for_Network => 0,
+		Redirect_for_Host => 1,
+		Redirect_for_Service_Network => 2,
+		Redirect_for_Service_Host => 3
+	) with Size => 8;
+
+	type CodeZero is range 0 .. 0 with Size => 8;
+	type Checksum is mod 2**16;
+	type Identifier is mod 2**16;
+	type Sequence_Number is mod 2**16;
+	type Pointer is mod 2**8;
+	type Timestamp is mod 2**32;
+	type Gateway_Inet_Addr is mod 2**32;
+	type Uint32Unused is range 0..0 with Size => 32;
+	type Uint24Unused is range 0..0 with Size => 24;
+
+	type Echo_Request_Reply is
+	   message
+			Tag : Tag;
+			Code : CodeZero;
+			Checksum : Checksum;
+			Identifier : Identifier;
+			Sequence_Number : Sequence_Number
+				then Data
+					with Length => Message'Last - Sequence_Number'Last;
+			Data : Opaque;
+	  end message;
+
+	type Destination_Unreachable is
+		message
+			Tag : Tag;
+			Code : CodeDestUnreachable;
+			Checksum : Checksum;
+			Unused : Uint32Unused
+				then Data
+					with Length => 224;
+			Data : Opaque;
+	  end message;
+
+	type Time_Exceeded is
+		message
+			Tag : Tag;
+			Code : CodeTimeExceeded;
+			Checksum : Checksum;
+			Unused : Uint32Unused
+				then Data
+					with Length => 224;
+			Data : Opaque;
+	  end message;
+
+	type Parameter_Problem is
+		message
+			Tag : Tag;
+			Code : CodeZero;
+			Checksum : Checksum;
+			Pointer : Pointer;
+			Unused : Uint24Unused
+				then Data
+					with Length => 224;
+			Data : Opaque;
+	  end message;
+
+	type Source_Quench is
+		message
+			Tag : Tag;
+			Code : CodeZero;
+			Checksum : Checksum;
+			Unused : Uint32Unused
+				then Data
+					with Length => 224;
+			Data : Opaque;
+	end message;
+
+	type Redirect is
+		message
+			Tag : Tag;
+			Code : CodeRedirect;
+			Checksum : Checksum;
+			Gateway_Inet_Addr : Gateway_Inet_Addr
+				then Data
+					with Length => 224;
+			Data : Opaque;
+	  end message;
+
+	type Timestamp_Reply is
+		message
+			Tag : Tag;
+			Code : CodeZero;
+			Checksum : Checksum;
+			Identifier : Identifier;
+			Sequence_Number : Sequence_Number;
+			Originate_Timestamp : Timestamp;
+			Receive_Timestamp : Timestamp;
+			Transmit_Timestamp : Timestamp;
+	  end message;
+
+	type Information_Request_Reply is
+		message
+			Tag : Tag;
+			Code : CodeZero;
+			Checksum : Checksum;
+			Identifier : Identifier;
+			Sequence_Number : Sequence_Number;
+	  end message;
```

Comment:
Please use an indentation level of 3.

Reply:
Somehow a tab character (with an indention of 3 in my ide) was used instead of 3 spaces.